### PR TITLE
test: migrate to vitest

### PR DESCRIPTION
**Pull request by okxiaoliang4**
migrate to `vitest`

`vitest` test for source
```shell
Test Files  52 passed (52)
     Tests  403 passed | 1 todo (404)
      Time  16.35s (in thread 4.93s, 331.63%)
```

`vitest` test for product
```shell
Test Files  52 passed (52)
     Tests  403 passed | 1 todo (404)
      Time  14.89s (in thread 4.58s, 325.36%)

```

`jest` test for source
```shell
Test Suites: 53 passed, 53 total
Tests:       1 skipped, 1 todo, 404 passed, 406 total
Snapshots:   4 passed, 4 total
Time:        38.9 s
Ran all test suites matching /tests\\/i.
```

`jest` test for product
```
Test Suites: 53 passed, 53 total
Tests:       1 skipped, 1 todo, 404 passed, 406 total
Snapshots:   4 passed, 4 total
Time:        37.815 s
Ran all test suites matching /tests\\/i.
```

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "allowSyntheticDefaultImports": true,
     "noUnusedLocals": true,
     "baseUrl": ".",
+    "types": ["vitest/globals"],
```
Comment (lmiller1990) [inline tsconfig.json:15]: TBH I much prefer explicit imports, eg `import { describe, it } from 'vitest'`.

**Comment by lmiller1990** [comment]
Cool - happy to merge this if we can get everything green. Definitely not my biggest priority, but nice to see Vitest is getting more mature.

**Review by okxiaoliang4 — COMMENTED** [review]
self reviewed :)

File: tests/features/async-components.spec.ts [hunk 2]
```diff
@@ -87,9 +93,7 @@ describe('defineAsyncComponent', () => {
     expect(wrapper.html()).toContain('Error Component')
   })
 
-  // TODO: Find out why this does not work
-  // Is it valid to have an AsyncComponent as the root? Was it ever?
-  it.skip('works when AsyncComponent is the root', async () => {
+  it('works when AsyncComponent is the root', async () => {
```
Comment (okxiaoliang4) [inline tests/features/async-components.spec.ts:96]: fix the test fail

**Review by cexbrayat — APPROVED** [review]
LGTM, thanks!

**Comment by cexbrayat** [comment]
@okxiaoliang4 Would you mind rebasing the PR and fixing the conflicts please? I'll merge when it's done

**Comment by lmiller1990** [comment]
🎉 

I actually have no problem with Jest... but happy to adopt Vitest, much of a muchness, really.

**Comment by okxiaoliang4** [comment]
@cexbrayat done.

**Pull request merged**